### PR TITLE
Upload file for `customization.logo.*` settings

### LINKED ISSUE #4426: Custom branding: PDF reports
## Description

As described in #4392, we need to add the ability to white-label the reports generated by the App.

Screenshot from 2022-08-31 12-08-17

### Pre-conditions
- [x] #4434

### Tasks
- [x] (1) Use of custom logo in reports
- [x] (2) #3892 
- [x] (3) Use of custom copyright mark in reports

### Definitions
- _wazuh_logo_: Wazuh's original logo, as seen in the image.
- _custom_logo_: user chosen logo. By default, this logo will be a copy of _wazuh_logo_.

### Strategy
These 2 logos will live in the app's space simultaneously (the paths are yet to be defined). The app will always use the _custom_logo_. The following states and mutations are allowed:

- (S) **initial**: _custom_logo_ = _wazuh_logo_
- (M) **update**: _custom_logo_ overwritten with uploaded logo
- (M) **delete**: _custom_logo_ overwritten with _wazuh_logo_

**Depends on**: #4443

### LINKED ISSUE #4434: Custom branding: assets upload
# Description

In order to ease the white-label process to the user, file upload for images and logos needs to be implemented.

## Tasks

- [ ] (1) Design and implement backend endpoints to allow read/write operations for custom branding
- [ ] (2) Adapt the UI to allow the upload of images and icons.

We won't let the user select the name or folder of the file, just upload one, or reset to the original.

The tasks above might be elevated to issues, for the time being, this is what it's required:

1. GET, PUT & DELETE operations to read, upload and reset custom imagery. No files are created or removed, only read and overwritten. See the parent issue for details.
We need to be very careful with these endpoints in terms of security. The image must be validated in the backend (format and dimensions). Their names and paths must be decided beforehand, by the backend.

2. Replace the text inputs by [file pickers](https://elastic.github.io/eui/#/forms/form-controls#file-pickers) from the EUI framework. Validate the files in the frontend (format and dimensions). Send the files to the new endpoints.
image

**Depends on**: #4443

### PR DESCRIPTION
**Pull request by Desvelao**
# Description

This PR improves and facilitates how the user can customize the logos through the ability to upload and remove files from `Settings/Configuration`. Moreover, it extends the setting description with properties related to files.

## Changes
- `customization.logo.*` settings definition

| Key                              | Description                                                   | Values  | Extensions                  | File size |
|----------------------------------|---------------------------------------------------------------|---------|-----------------------------|-----------|
| `customization.logo.app`         | Set the custom logo for the menu and other sections           | String  | `jpeg`, `jpg`, `png`, `svg` | 1 MB      |
| `customization.logo.healthcheck` | Set the custom logo for the app's health check                | String  | `jpeg`, `jpg`, `png`, `svg` | 1 MB      |
| `customization.logo.reports`     | Set the custom logo for the PDF reports                       | String  | `jpeg`, `jpg`, `png`        | 1 MB      |
| `customization.logo.sidebar`     | Set the custom logo for the category in the navigation drawer | String. | `jpeg`, `jpg`, `png`, `svg` | 1 MB      |

- Add endpoints to manage the file upload/deletion
  - Upload: `PUT /utils/configuration/files/{key}`
    - Maximum request body is `1MB`
    - File extensions and request body size restrictions
    - File extension is detected through the file buffer. The `.jpeg` files will be saved as `.jpg`.
    - Store the file in the file system and save the setting with a relative URL to the exposed asset.
    - The `customization.logo.*` files are stored in:
      - file system: `<WAZUH_PLUGIN>/public/assets/custom/images/<key><extension>`
      - exposed URL: `<PLUGIN_PLATFORM_ADDRESS>/plugins/wazuh/assets/custom/images/<key><extension>`
  - Deletion: `DELETE /utils/configuration/files/{key}`
    - Remove the file from the file system
    - Set the setting to a empty string `""`
- Add plugin setting type: `filepicker`
- Add file picker form input component
- If some `customization.logo.*` is set, then:
  - Display the customized image in `Settings/Configuration`
  - Add a button to remove the customized image
- Create a function to convert bytes to a meaningful unit.
- Extend the setting description with:
  - Supported file extensions
  - Recommended image dimensions
  - File size restrictions

- Tests
  - Add test for the `InputForm` component of type `filepicker`
  - Add test for the new endpoints:
    - `PUT /utils/configuration/files/{key`
      - Validate the received file: max size
    - `DELETE /utils/configuration/files/{key}`
      - Validate the setting: known and unknown

Closes #4434 

**Screenshots**

- Setting configuration using a file picker
image
- When there is a custom image set
image
- Hovering the remove icon
image
- Confirmation to remove the custom image
image

# Tests

## Manual tests

**Scenario**: Select a file with the suggested extension for each `customization.logo.*` setting in `Settings/Configuration`
  **When** the user drag-and-drops or selects a file in the file picker of `Settings/Configuration`
  **Then** the file picker should suggest files with the supported extensions

> This scenario should be done with all the `customization.logo.*` settings.

  
**Scenario**: Upload the configuration for a `customization.logo.*` setting in `Settings/Configuration`
  **When** the user navigates to `Settings/Configuration`, selects a valid file and clicks on the `Save` button
  **Then** the file should be saved in the expected path of the file system: `WAZUH_PLUGIN_PATH/public/assets/custom/images` whose filename is the setting key and the extension is the same as the uploaded file. (`jpeg` files will be saved as `jpg`)
  **And** the image should be displayed on the UI depending on the setting customized. Keep in mind that currently, the `customization.logo.sidebar` requires reloading the browser tab.
  **And** the configuration file should be modified to include the setting and its value is a relative URL that should be exposed.

> This scenario should be done with all the `customization.logo.*` settings.

**Scenario**: Select a file that exceeds the size limit displays a validation error in `Settings/Configuration`
  **When** the user navigates to `Settings/Configuration` and selects a file that exceeds the size limit. Above 1 MB.
  **Then** a validation error should be displayed
  
> This scenario should be done with all the `customization.logo.*` settings.

**Scenario**: Delete the customization of each `customization.logo.*` setting in `Settings/Configuration`
  **Given** the setting is customized
  **When** the user navigates to `Settings/Configuration` and clicks on the button to remove the file
  **Then** the file should be removed from the file system: `WAZUH_PLUGIN_PATH/public/assets/custom/images`
  **And** the UI should display the default image. Keep in mind that currently the `customization.logo.sidebar` setting requires reloading the browser tab
  **And** the configuration file for the setting should be cleaned. This means that its value is an empty string `""`
  
> This scenario should be done with all the `customization.logo.*` settings.

  
**Scenario**: Update the plugin configuration from `Settings/Configuration` by modifying multiple settings at the same time
  **Given** the user navigates to `Settings/Configuration` and modifies some setting that is not a `customization.logo.*` and one of `customization.logo.*` setting
  **When** the user clicks on the button to save
  **Then** the file should be saved in the expected path of the file system: `WAZUH_PLUGIN_PATH/public/assets/custom/images` whose filename is the setting key and the extension is the same as the uploaded file.  (`jpeg` files will be saved as `jpg`)
  **And** the image should be displayed depending on the setting customized. Currently, the `customization.logo.sidebar` requires reloading the browser tab
  **And** the configuration file should be modified to include the setting and its value is a relative URL that should be exposed
  **And** all the modified settings should be saved in the configuration file
  
**Scenario**: Update the plugin configuration from `Settings/Configuration` modifying multiple settings at the same time with 2 or more `customization.logo.*` settings
  **Given** the user navigates to `Settings/Configuration` and modifies some setting that is not a `customization.logo.*` and 2 or more of `customization.logo.*` settings
  **When** the user clicks on the button to save
  **Then** the files should be saved in the expected path of the file system: `WAZUH_PLUGIN_PATH/public/assets/custom/images` whose filename is the setting key and the extension is the same as the uploaded file. (`jpeg` files will be saved as `jpg`)
  **And** the images should be displayed depending on the setting customized. Keep in mind that currently, the `customization.logo.sidebar` requires reloading the browser tab.
  **And** the configuration should be modified to include the settings and its value is a relative URL that should be exposed.
  **And** all the modified settings should be saved in the configuration file
  
**Scenario**: Select a valid file for each `customization.logo.*` setting in `Settings/Configuration` with different aspect ratio
  **When** the user navigates to `Settings/Configuration` and drag-and-drops or selects a file
  **Then** the file picker should suggest the supported file extensions
  **And** renders properly the uploaded custom logo, without visual glitches or styling issues

> This scenario should be done with all the `customization.logo.*` settings.
  
## Automatic tests
- `common/services/file-size.test.ts`
- `public/components/common/form/index.test.tsx`
- `server/routes/wazuh-utils/wazuh-utils.test.ts`
- `common/plugin-settings.test.ts`
- `common/services/file-extension.test.ts`
  
**Run all PR tests**
```shell
yarn test:jest common/services/file-size.test.ts public/components/common/form/index.test.tsx server/routes/wazuh-utils/wazuh-utils.test.ts common/plugin-settings.test.ts common/services/file-extension.test.ts
```

### CODE REVIEW

**Comment by asteriscos** [comment]
### Different aspect ratio example

image

**Comment by asteriscos** [comment]
## Pending

- [ ] Clear the file picker state when the file is uploaded

image

**Comment by Desvelao** [comment]
# Update

- Update the development branch with the last changes of the base branch.

**Comment by Desvelao** [comment]
# Requirements
Yesterday, in a sync meeting, we decide to get the file extension from the buffer instead of the current logic that send the file and the extension as another field.

**Comment by Desvelao** [comment]
# Update
- Changed as the file extension was gotten when uploading the file. Now, it gets from the file buffer.
  - Create a service to get the file extension
  - Add some tests for the service
  - Remove the previous logic and adapt to use the service

**Comment by Desvelao** [comment]
# Update
- Update the development branch with the base branch

**Review by yenienserrano — APPROVED** [review]
LGTM

File: common/constants.ts [hunk 1]
```diff
@@ -342,6 +342,11 @@ export const DOCUMENTATION_WEB_BASE_URL = "https://documentation.wazuh.com";
 // Default Elasticsearch user name context
 export const ELASTIC_NAME = 'elastic';
 
+
+// Customization
```
Comment (AlexRuiz7) [inline common/constants.ts:346]: Default 1 MB (actual value is in bytes)

**Comment by Desvelao** [comment]
> Would you have to change to the branch you are merging to?

Changed.

**Review by AlexRuiz7 — CHANGES_REQUESTED** [review]

File: public/components/settings/configuration/components/categories/components/show-toasts.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	EuiFlexItem,
+} from '@elastic/eui';
+import { PLUGIN_PLATFORM_NAME } from '../../../../../../../common/constants';
+import _ from 'lodash';
+import { getToasts } from '../../../../../../kibana-services';
+
+export const toastRequiresReloadingBrowserTab = () => {
+	getToasts().add({
+	  color: 'success',
+	  title: 'This setting requires you to reload the page to take effect.',
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/components/categories/components/show-toasts.tsx:14]: The messages were not changed, they only were moved to another file, buy I will change the message with your suggestion.

File: public/components/settings/configuration/components/categories/components/show-toasts.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+	EuiButton,
+	EuiFlexGroup,
+	EuiFlexItem,
+} from '@elastic/eui';
+import { PLUGIN_PLATFORM_NAME } from '../../../../../../../common/constants';
+import _ from 'lodash';
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/components/categories/components/show-toasts.tsx]: Removed.

File: public/components/settings/configuration/components/categories/components/show-toasts.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/components/categories/components/show-toasts.tsx]: Changed.

File: public/components/settings/configuration/components/categories/components/show-toasts.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		  <EuiButton onClick={() => window.location.reload()} size="s">Reload page</EuiButton>
+		</EuiFlexItem>
+	  </EuiFlexGroup>
+	});
+  };
+  
+export const toastRequiresRunningHealthcheck = () => {
+	const toast = getToasts().add({
+	  color: 'warning',
+	  title: 'You must execute the health check for the changes to take effect',
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/components/categories/components/show-toasts.tsx]: Changed.

File: public/components/settings/configuration/configuration.tsx [hunk 5]
```diff
@@ -29,15 +29,18 @@ import {
 import store from '../../../redux/store'
 import { updateSelectedSettingsSection } from '../../../redux/actions/appStateActions';
 import { withUserAuthorizationPrompt, withErrorBoundary, withReduxProvider } from '../../common/hocs';
-import { PLUGIN_PLATFORM_NAME, PLUGIN_SETTINGS, PLUGIN_SETTINGS_CATEGORIES, UI_LOGGER_LEVELS, WAZUH_ROLE_ADMINISTRATOR_NAME } from '../../../../common/constants';
+import { EpluginSettingType, PLUGIN_PLATFORM_NAME, PLUGIN_SETTINGS, PLUGIN_SETTINGS_CATEGORIES, UI_LOGGER_LEVELS, WAZUH_ROLE_ADMINISTRATOR_NAME } from '../../../../common/constants';
 import { compose } from 'redux';
-import { getSettingsDefaultList, groupSettingsByCategory, getCategorySettingByTitle } from '../../../../common/services/settings';
+import { getPluginSettingDescription, getSettingsDefaultList, groupSettingsByCategory, getCategorySettingByTitle } from '../../../../common/services/settings';
+import _ from 'lodash';
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/configuration.tsx]: Removed.

File: public/components/settings/configuration/configuration.tsx [hunk 6]
```diff
@@ -29,15 +29,18 @@ import {
 import store from '../../../redux/store'
 import { updateSelectedSettingsSection } from '../../../redux/actions/appStateActions';
 import { withUserAuthorizationPrompt, withErrorBoundary, withReduxProvider } from '../../common/hocs';
-import { PLUGIN_PLATFORM_NAME, PLUGIN_SETTINGS, PLUGIN_SETTINGS_CATEGORIES, UI_LOGGER_LEVELS, WAZUH_ROLE_ADMINISTRATOR_NAME } from '../../../../common/constants';
+import { EpluginSettingType, PLUGIN_PLATFORM_NAME, PLUGIN_SETTINGS, PLUGIN_SETTINGS_CATEGORIES, UI_LOGGER_LEVELS, WAZUH_ROLE_ADMINISTRATOR_NAME } from '../../../../common/constants';
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/configuration.tsx]: Removed.

File: public/components/settings/configuration/configuration.tsx [hunk 7]
```diff
@@ -154,8 +182,29 @@ const WzConfigurationSettingsProvider = (props) => {
...
+              // methods used to remove the files. Of this way, we can skip the errors
+              // due to missing methods.
+              // This workaround is based in @elastic/eui v29.3.2
+              // https://github.com/elastic/eui/blob/v29.3.2/src/components/form/file_picker/file_picker.tsx#L107-L108
+              {stopPropagation: () => {}, preventDefault: () => {}}
+            );
+          }catch(error){
+            // TODO: There is an error related to accessing to `target` of undefined
+            // that we should review it.
+          }
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/configuration.tsx]: It couldn't replicate it. Maybe some previous change doesn't cause this problem anymore.

**Comment by Desvelao** [comment]
# Update
- Fixed a problem with an unexpected attribute for the input of type number.
- Fixed a problem when removing the configuration for `customization.logo.reports` that displayed the original logo when should not.

File: public/components/settings/configuration/components/categories/components/show-toasts.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+			}} size="s">Execute health check</EuiButton>
+		  </EuiFlexItem>
+		</EuiFlexGroup>
+	});
+  };
+  
+export const toastRequiresRestartingPluginPlatform = () => {
+	getToasts().add({
+	  color: 'warning',
+	  title: `You must restart ${PLUGIN_PLATFORM_NAME} for the changes to take effect`,
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/components/categories/components/show-toasts.tsx]: Changed.

File: public/components/settings/configuration/configuration.tsx [hunk 9]
```diff
@@ -29,15 +29,18 @@ import {
...
+import { EpluginSettingType, PLUGIN_PLATFORM_NAME, PLUGIN_SETTINGS, PLUGIN_SETTINGS_CATEGORIES, UI_LOGGER_LEVELS, WAZUH_ROLE_ADMINISTRATOR_NAME } from '../../../../common/constants';
 import { compose } from 'redux';
-import { getSettingsDefaultList, groupSettingsByCategory, getCategorySettingByTitle } from '../../../../common/services/settings';
+import { getPluginSettingDescription, getSettingsDefaultList, groupSettingsByCategory, getCategorySettingByTitle } from '../../../../common/services/settings';
+import _ from 'lodash';
 import { Category } from './components/categories/components';
 import { WzRequest } from '../../../react-services';
 import { UIErrorLog, UIErrorSeverity, UILogLevel, UI_ERROR_SEVERITIES } from '../../../react-services/error-orchestrator/types';
 import { getErrorOrchestrator } from '../../../react-services/common-services';
 import { getToasts } from '../../../kibana-services';
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/settings/configuration/configuration.tsx:39]: Removed.

File: common/services/file-extension.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export function getFileExtensionFromBuffer(buffer: Buffer): string {
+	const imageFormat = buffer.toString('hex').substring(0, 4);
+	switch (imageFormat) {
+		case '4749':
+			return 'gif';
+		case 'ffd8':
+			return 'jpg'; // Also jpeg
+		case '8950':
+			return 'png';
+		case '3c3f':
```
Comment (AlexRuiz7) [inline common/services/file-extension.ts:17]: I did some research and looks like `3c73` is also a valid file signature for SVG files.

Lots of examples [here](https://www.w3schools.com/graphics/svg_examples.asp). Copy-paste and remove the HTML tags:

```xml
<svg height="400" width="450">
<path id="lineAB" d="M 100 350 l 150 -300" stroke="red" stroke-width="3" fill="none" />
  <path id="lineBC" d="M 250 50 l 150 300" stroke="red" stroke-width="3" fill="none" />
  <path d="M 175 200 l 150 0" stroke="green" stroke-width="3" fill="none" />
  <path d="M 100 350 q 150 -300 300 0" stroke="blue" stroke-width="5" fill="none" />
  <!-- Mark relevant points -->
  <g stroke="black" stroke-width="3" fill="black">
    <circle id="pointA" cx="100" cy="350" r="3" />
    <circle id="pointB" cx="250" cy="50" r="3" />
    <circle id="pointC" cx="400" cy="350" r="3" />
  </g>
  <!-- Label the points -->
  <g font-size="30" font-family="sans-serif" fill="black" stroke="none" text-anchor="middle">
    <text x="100" y="350" dx="-30">A</text>
    <text x="250" y="50" dy="-10">B</text>
    <text x="400" y="350" dx="30">C</text>
  </g>
  Sorry, your browser does not support inline SVG.
</svg>
```

image

**Review by AlexRuiz7 — CHANGES_REQUESTED** [review]
Errors during testing:

- Delete file: 400 Bad Request
Screenshot from 2022-10-26 16-17-18


- Upload valid SVG file: 400 Bad Request
image

**Review by Mayons95 — COMMENTED** [review]
TR: 
<details>
[Screencast from 26-10-22 10:34:07.webm](https://user-images.githubusercontent.com/104914131/198067322-c363fa7c-7d62-4fc6-974c-4b5dd74f9345.webm)


[Screencast from 26-10-22 10:49:12.webm](https://user-images.githubusercontent.com/104914131/198067359-6f4c38fd-87d7-46d3-8cbe-53edb79ae8b7.webm)




</details>

**Comment by Desvelao** [comment]
# Update
- Change the layout of the filepicker input and the logo with the button to remove the configuration.

**Screenshots**
:computer: Desktop
image
image

:iphone: Mobile
image
image

**Comment by Mayons95** [comment]
TR: :heavy_check_mark: 

<details>

[Screencast from 26-10-22 13:57:01.webm](https://user-images.githubusercontent.com/104914131/198089786-a29fb484-982c-42e0-ad0f-c82f19a8010d.webm)

[Screencast from 26-10-22 13:58:09.webm](https://user-images.githubusercontent.com/104914131/198089828-6eacda3b-6d4d-49dc-89bd-95aedecb8795.webm)


</details >

**Review by Mayons95 — DISMISSED** [review]
LGTM

**Review by yenienserrano — APPROVED** [review]
LGTM

**Review by yenienserrano — DISMISSED** [review]

**Comment by Mayons95** [comment]
TR Comments:
We performed a script with @Desvelao and @asteriscos to test the API file upload, I left the script here, to use it, it's necessary to update the cookie value.

[script.tar.gz](https://github.com/wazuh/wazuh-kibana-app/files/9879764/script.tar.gz)

**Review by Mayons95 — APPROVED** [review]
TR: :heavy_check_mark: 
LGTM

**Comment by Desvelao** [comment]
Some problems found by @AlexRuiz7 

**Environment: Kibana 7.10.2 development mode.**

## The endpoint to remove the configuration for the `customization.logo.*` setting doesn't work.
Endpoint: `DELETE /utils/configuration/files/{key}`

We were researching and saw that could be caused by the development proxy (not using the `--no-base-path` when starting the server) and accessing using the `5601` port instead of the mentioned port `5603` in the logs of the platform. We do not know the reason for the problem.

Others requests with the same HTTP verb (`DELETE`) fail too.
image

## Some custom logos are not visible when using SVG images.
This problem happens in:
- custom image preview in `Settings/Configuration`
- plugin menu

This is caused because the SVG image has not defined the `height` and `width` properties.
image

If these properties are added, the problem disappears.
image
image

@AlexRuiz7 reported that by using the same SVG for other settings as `customization.logo.healthcheck` and `customization.logo.sidebar`, the images are displayed correctly. This means that some styles could be affecting how the browser should render the image.

### Research - Custom image preview in `Settings/Configuration`

I found that if the `width: auto` CSS property is removed from the element style, the image is displayed correctly. This property is set by the `EuiImage` component when the `size` property is defined as a number https://github.com/elastic/eui/blob/v29.3.2/src/components/image/image.tsx#L140.

EDIT: I pushed a commit to solve this problem.

### Research -  Plugin menu

EDIT: I pushed a commit to solve this problem.

**Comment by Desvelao** [comment]
In a meeting with @asteriscos , we saw a problem related to the browser cache and the `customization.logo.*` settings used in the frontend UI.

How to replicate:
1. Upload an image with some extension (`jpg`, `png`, `svg`)
2. Upload another image with the same extension as step 1. The UI display doesn't change and displays the image of step one. Cleaning the browser cache solves the problem.

We want to research a solution that does not implicate cleaning the browser cache and minimizes the users.

**Comment by Desvelao** [comment]
> In a meeting with @asteriscos , we saw a problem related to the browser cache and the `customization.logo.*` settings used in the frontend UI.
> 
> How to replicate:
> 
> 1. Upload an image with some extension (`jpg`, `png`, `svg`)
> 2. Upload another image with the same extension as step 1. The UI display doesn't change and displays the image of step one. Cleaning the browser cache solves the problem.
> 
> We want to research a solution that does not implicate cleaning the browser cache and minimizes the users.

Solved in https://github.com/wazuh/wazuh-kibana-app/pull/4504/commits/ccd57ca7c2d00ec43eb199f452ddf5a1b42b5188.

**Comment by asteriscos** [comment]
> > In a meeting with @asteriscos , we saw a problem related to the browser cache and the `customization.logo.*` settings used in the frontend UI.
> > How to replicate:
> > 
> > 1. Upload an image with some extension (`jpg`, `png`, `svg`)
> > 2. Upload another image with the same extension as step 1. The UI display doesn't change and displays the image of step one. Cleaning the browser cache solves the problem.
> > 
> > We want to research a solution that does not implicate cleaning the browser cache and minimizes the users.
> 
> Solved in [ccd57ca](https://github.com/wazuh/wazuh-kibana-app/commit/ccd57ca7c2d00ec43eb199f452ddf5a1b42b5188).

Now when an image is uploaded, the path saved in the configuration includes a version number to handle browser cache issues.
`customization.logo.app: "custom/images/customization.logo.app.png?v=1666948554416"`

**Review by AlexRuiz7 — APPROVED** [review]
- Code review: ✔️ 
- Test: ✔️

**Review by Mayons95 — APPROVED** [review]
LGTM 
TR: :heavy_check_mark: 
Test with the last implemented changes

**Pull request merged**